src/main.rs

```diff
@@ -17,6 +17,13 @@ enum Opts {
         #[structopt(short = "r", long = "remote", help = "Remote ssh build server")]
         remote: Option<String>,
 
+        #[structopt(
+            short = "p",
+            long = "port",
+            help = "Custom port for ssh on the build server"
+        )]
+        port: Option<String>,
```

Comment:
This should be `Option<u16>`, a `String` allows too many invalid inputs. This would also make `port` `Copy`, avoiding al these `clone()`s.